src/ai/llm/messages/manager.ts

```diff
@@ -54,30 +54,65 @@ export class MessageManager {
      * Creates a new MessageManager instance
      *
      * @param formatter Formatter implementation for the target LLM provider
-     * @param systemPrompt Optional system prompt to initialize the conversation
+     * @param systemPrompt System prompt for the conversation
      * @param maxTokens Maximum token limit for the conversation history. Triggers compression if exceeded and a tokenizer is provided.
      * @param tokenizer Tokenizer implementation used for counting tokens and enabling compression.
      * @param compressionStrategies Optional array of compression strategies to apply sequentially when maxTokens is exceeded. Defaults to [MiddleRemoval, OldestRemoval]. Order matters.
      */
     constructor(
         formatter: IMessageFormatter,
-        systemPrompt: string | null = null,
-        maxTokens: number | null = null,
-        tokenizer: ITokenizer | null = null,
+        systemPrompt: string,
+        maxTokens: number,
+        tokenizer: ITokenizer,
         compressionStrategies: ICompressionStrategy[] = [
             new MiddleRemovalStrategy(),
             new OldestRemovalStrategy(),
         ]
     ) {
+        if (!systemPrompt) throw new Error('systemPrompt is required');
+        if (maxTokens == null) throw new Error('maxTokens is required');
+        if (!tokenizer) throw new Error('tokenizer is required');
         this.formatter = formatter;
-        if (systemPrompt) {
-            this.setSystemPrompt(systemPrompt);
-        }
+        this.systemPrompt = systemPrompt;
         this.maxTokens = maxTokens;
         this.tokenizer = tokenizer;
         this.compressionStrategies = compressionStrategies;
     }
 
+    /**
+     * Returns the current token count of the conversation history.
+     * @returns The number of tokens in the current history
+     */
+    getTokenCount(): number {
```

Comment:
added this back for llmservice to have a way to get current token count without needing the tokenizer/formatter